tests/functional/test_install_upgrade.py

```diff
@@ -37,8 +38,8 @@ def test_invalid_upgrade_strategy_causes_error(script):
 
 
 def test_only_if_needed_does_not_upgrade_deps_when_satisfied(
-    script, resolver_variant, with_wheel
-):
+    script: PipTestEnvironment, resolver_variant: str, with_wheel: None
+) -> None:
```

Comment:
Wondering whether it’d be worthwhile to have a type alias `ResolverVariant = Literal["resolvelib", "legacy"]` instead.

Reply:
Also it’s probably better to move `with_wheel` to `pytest.mark.usefixtures` instead. This eliminates the awkward unused `None` argument.

Reply:
> Wondering whether it’d be worthwhile to have a type alias ResolverVariant = Literal["resolvelib", "legacy"] instead.

:+1: Done in latest.

> Also it’s probably better to move with_wheel to pytest.mark.usefixtures instead. This eliminates the awkward unused None argument.

Thanks for sharing this technique. I applied it in #10492